src/client/upload.rs

```diff
@@ -0,0 +1,51 @@
+use std::{
+    fs::File,
+    io::{BufReader, Read},
+    path::PathBuf,
+    str::FromStr,
+};
+
+use crate::{
+    consts::VERSION,
+    currency::{arweave::Arweave, ethereum::Ethereum, solana::Solana, Currency, CurrencyType},
+    error::BundlrError,
+    tags::Tag,
+    Bundlr,
+};
+use reqwest::Url;
+
+pub async fn run_upload(
+    file_path: String,
+    url: Url,
+    wallet: &str,
+    currency: CurrencyType,
+) -> Result<String, BundlrError> {
+    let currency: Box<dyn Currency> = match currency {
+        CurrencyType::Arweave => {
+            let wallet = PathBuf::from_str(&wallet).expect("Invalid wallet path");
+            Box::new(Arweave::new(wallet, None))
+        }
+        CurrencyType::Solana => Box::new(Solana::new(wallet, None)),
+        CurrencyType::Ethereum => Box::new(Ethereum::new(wallet, None)),
+        CurrencyType::Erc20 => todo!(),
+        CurrencyType::Cosmos => todo!(),
+    };
+    let f = File::open(file_path.clone()).expect("Invalid file path");
+    let mut reader = BufReader::new(f);
+    let mut buffer = Vec::new();
+
+    // Read file into vector.
+    reader.read_to_end(&mut buffer)?;
+
+    let base_tag = Tag::new("User-Agent", &format!("bundlr-sdk-rs/{}", VERSION));
+
+    // Read.
+    let bundlr = Bundlr::new(url, currency.as_ref()).await;
```

Comment:
Is there need to support multiple currency types with single `Bundlr` instance? If not, then you could add currency as a generic parameter to `Bundlr` and you could get rid of boxing the currency. This is probably bigger change, so maybe a future work item. Once you've read this, just resolve ticket, I'm not expecting this change with this PR, just bringing up the question and idea for future development.